#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/neuronics/ipr/protos/IprHd6m180.proto"

WorldInfo {
  info [
    "A demo world using the IPR HD6M180"
  ]
  title "IPR Cube"
  basicTimeStep 20
  contactProperties [
    ContactProperties {
      material1 "target"
      material2 "gripper"
      coulombFriction [
        2
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5754251625887666 0.5775870732849231 0.5790328617909769 2.088622597672399
  position 0.2873127870002446 -0.09470727083186935 5.324834538962031
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF FLOOR Solid {
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.901961 0.901961 0.980392
        roughness 1
        metalness 0
      }
      geometry DEF PLANE Plane {
        size 5 5
      }
    }
  ]
  name "ground"
  boundingObject DEF PLANE Plane {
    size 5 5
  }
  locked TRUE
}
Solid {
  translation 0.4 0.2 0.71
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.3 0.15
      }
    }
  ]
  name "box"
  boundingObject Box {
    size 0.2 0.3 0.15
  }
}
Solid {
  translation 0.18 -0.18 0.78
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.3
      }
    }
  ]
  name "box(3)"
  boundingObject Box {
    size 0.1 0.1 0.3
  }
}
Solid {
  translation -0.19 -0.02 0.78
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.3
      }
    }
  ]
  name "box(2)"
  boundingObject Box {
    size 0.1 0.1 0.3
  }
}
Solid {
  translation -0.04 -0.45 0.67
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.3 0.08
      }
    }
  ]
  name "box(1)"
  boundingObject Box {
    size 0.2 0.3 0.08
  }
}
DEF TARGET Solid {
  translation 0.371547479355474 0.20779376287061257 0.8109980380000001
  rotation 0.6222785198179925 0.7827959145097324 -3.891479216458222e-17 3.141592653589793
  children [
    DEF CUBE Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.365482152
        metalness 0
      }
      geometry Box {
        size 0.053 0.052 0.052
      }
    }
  ]
  name "cube0"
  contactMaterial "target"
  boundingObject DEF BOUNDING_CUBE Box {
    size 0.053 0.052 0.052
  }
  physics DEF PHYSICS_CUBE Physics {
    density -1
    mass 10
  }
}
DEF TARGET Solid {
  translation -0.001155560000000011 0.292358 0.6589980379999998
  rotation 5.231648541983837e-17 -2.9426484612292056e-17 1 1.5410629278111594
  children [
    DEF CUBE Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 0.365482152
        metalness 0
      }
      geometry Box {
        size 0.053 0.052 0.052
      }
    }
  ]
  name "cube0(1)"
  contactMaterial "target"
  boundingObject DEF BOUNDING_CUBE Box {
    size 0.053 0.052 0.052
  }
  physics DEF PHYSICS_CUBE Physics {
    density -1
    mass 10
  }
}
DEF TARGET Solid {
  translation -0.022858199999999995 -0.437321 0.7359980379999999
  rotation 9.491596575198066e-18 6.020466367994512e-17 1 1.5410629278111594
  children [
    DEF CUBE Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 0.365482152
        metalness 0
      }
      geometry Box {
        size 0.053 0.052 0.052
      }
    }
  ]
  name "cube0(2)"
  contactMaterial "target"
  boundingObject DEF BOUNDING_CUBE Box {
    size 0.053 0.052 0.052
  }
  physics DEF PHYSICS_CUBE Physics {
    density -1
    mass 10
  }
}
IprHd6m180 {
  hidden position_0_0 3.0099999999987164
  hidden rotation_2 0 0 1 3.138854999998708
  hidden position_2_0 -0.5499999992751028
  hidden rotation_3 -1 0 0 0.025555000724866408
  hidden position_3_0 2.149999935776324
  hidden rotation_4 0.9999999999999999 0 0 0.061250064223670395
  hidden position_4_0 -2.000000011599271
  hidden rotation_5 -1 0 0 0.025409988400689355
  hidden position_5_0 -2.899999999987425
  hidden rotation_6 0 0 1 0.09437000001257538
  hidden position_6_0 1.683848456202582e-08
  hidden position_6_1 1.8087761288754223e-08
  hidden translation_19 0.0057492841956934245 1.5420696134849782e-05 0.032376686761237616
  hidden rotation_19 -1.1784786577112836e-10 1 5.403176140763547e-11 0.19200000003173556
  hidden translation_20 -5.581862542441496e-05 -1.2441738605980385e-05 0.08385747171831093
  hidden rotation_20 -0.2988617141722241 0.9148236603697688 0.2716235376945406 2.0446332214544168e-11
  hidden translation_42 -0.007803435348028465 2.5096504455000834e-05 0.03227891805479843
  hidden rotation_42 -4.683391001821052e-12 -1 1.4567218729341646e-11 -2.949685307303199
  hidden translation_43 9.600335430662798e-05 -8.01343251501628e-06 0.08398925981086958
  hidden rotation_43 -2.0857551819197675e-12 -1 3.651637156298342e-12 3.1414999999395556
  translation 0 0 0.645
  controller "ipr_controller"
  supervisor TRUE
  tableSlot [
    Solid {
      translation -0.06 -0.61 0
      children [
        DEF PLANE Pose {
          translation 0.2225 0.5225 -0.027
          children [
            Shape {
              appearance DEF WOOD PBRAppearance {
                baseColor 0.839216 0.619608 0.0588235
                roughness 0.23459205900000013
                metalness 0
              }
              geometry Box {
                size 1 1.2 0.03
              }
            }
          ]
        }
        DEF LEG1 Pose {
          translation 0.03 0.1 -0.342
          children [
            DEF LEG Shape {
              appearance USE WOOD
              geometry Box {
                size 0.05 0.05 0.6
              }
            }
          ]
        }
        DEF LEG2 Pose {
          translation 0.03 1 -0.342
          children [
            USE LEG
          ]
        }
        DEF LEG3 Pose {
          translation 0.4 0.1 -0.342
          children [
            USE LEG
          ]
        }
        DEF LEG4 Pose {
          translation 0.4 1 -0.342
          children [
            USE LEG
          ]
        }
      ]
      boundingObject Group {
        children [
          USE PLANE
          USE LEG1
          USE LEG2
          USE LEG3
          USE LEG4
        ]
      }
    }
  ]
  wristSlot [
    DEF WRIST Solid {
    }
  ]
  leftGripperSlot [
    DistanceSensor {
      translation 0.025 0 0.07
      rotation 0 1 0 -1.047195307179586
      name "DS3"
      type "laser"
    }
    DistanceSensor {
      translation 0.017 0 0.09
      name "DS2"
      type "laser"
    }
    DistanceSensor {
      translation 0.01 0 0.1
      rotation 0 1 0 -1.5707953071795862
      name "DS1"
      rotationStep 0.261809
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
      type "laser"
      aperture 1.5724
    }
    Camera {
      translation 0.0336602 0 0.0663398
      rotation 0 1 0 -1.047195307179586
      name "cameraG2"
      boundingObject Box {
        size 0.01 0.01 0.01
      }
      physics Physics {
      }
    }
    TouchSensor {
      translation 0.00799994 0.00800001 0.10166
      rotation -0.25056306820569746 -0.935113254546187 0.25056206820543553 1.6378400000000604
      name "touch_sensorG1"
      boundingObject Box {
        size 0.035 0.02 0.02
      }
      physics Physics {
      }
    }
    Camera {
      translation 0 0.01 0.1
      rotation -0.1294279597735375 -0.9831056944488314 0.1294279597735375 1.58783
      name "cameraG1"
      boundingObject Box {
        size 0.01 0.01 0.01
      }
      physics Physics {
      }
    }
  ]
  rightGripperSlot [
    DistanceSensor {
      translation 0 0 -0.07
      rotation 0 0 1 -1.5707953071795862
      name "DSO2"
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0 0 -0.07
      rotation 0 0 1 3.14159
      name "DSO3"
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0 0 -0.07
      rotation 0 0 -1 -1.5707953071795862
      name "DSO4"
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0 0 -0.07
      name "DSO1"
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
      type "laser"
    }
    TouchSensor {
      translation 0 0 -0.19
      name "touch_sensorO1"
      boundingObject Box {
        size 0.14 0.13 0.3
      }
    }
    DistanceSensor {
      translation 0 0 0.08
      rotation 0 0 1 -1.5707953071795862
      name "DS10"
      type "laser"
    }
    DistanceSensor {
      translation 0 0 0.05
      rotation 0 0 1 -1.5707953071795862
      name "DS9"
      type "laser"
    }
    DistanceSensor {
      translation 0 0 0.08
      rotation 0 0 1 1.5708
      name "DS8"
      type "laser"
    }
    DistanceSensor {
      translation 0 0 0.05
      rotation 0 0 1 1.5708
      name "DS7"
      type "laser"
    }
    DistanceSensor {
      translation -0.025 0 0.07
      rotation 0 1 0 -2.094395307179586
      name "DS6"
      type "laser"
    }
    DistanceSensor {
      translation -0.018 0 0.09
      rotation 0 1 0 3.14159
      name "DS5"
      type "laser"
    }
    DistanceSensor {
      translation -0.0099999 0 0.1
      rotation 0 -0.9999999999999999 0 1.5707953071795862
      name "DS4"
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
      type "laser"
    }
    Camera {
      translation -0.035 0 0.065
      rotation 0 1 0 -2.094395307179586
      name "cameraG3"
      boundingObject Box {
        size 0.01 0.01 0.01
      }
      physics Physics {
      }
    }
    GPS {
      translation -0.004 0 0.1
      rotation 0 1 0 -1.5707953071795862
      boundingObject Box {
        size 0.005 0.005 0.005
      }
      physics Physics {
      }
    }
    TouchSensor {
      translation -0.00832225 0.0123469 0.101194
      rotation -0.18218305171148239 -0.9615182729207506 0.20565005837244063 1.63782
      name "touch_sensorG2"
      boundingObject Box {
        size 0.035 0.02 0.02
      }
      physics Physics {
      }
    }
  ]
}
